exporter/awsxrayexporter/internal/translator/http.go

```diff
@@ -33,62 +47,63 @@ func makeHTTP(span ptrace.Span) (map[string]pcommon.Value, *awsxray.HTTPData) {
 			hasHTTP = true
-		case conventions.AttributeHTTPClientIP:
+		case conventions.AttributeHTTPClientIP, AttributeClientAddress:
+			//urlParts[conventions.AttributeHTTPClientIP] = value.Str()
 			info.Request.ClientIP = awsxray.String(value.Str())
 			info.Request.XForwardedFor = aws.Bool(true)
 			hasHTTP = true
-		case conventions.AttributeHTTPUserAgent:
+		case conventions.AttributeHTTPUserAgent, AttributeUserAgentOriginal:
 			info.Request.UserAgent = awsxray.String(value.Str())
 			hasHTTP = true
-		case conventions.AttributeHTTPStatusCode:
+		case conventions.AttributeHTTPStatusCode, AttributeHTTPResponseStatusCode:
 			info.Response.Status = aws.Int64(value.Int())
 			hasHTTP = true
-		case conventions.AttributeHTTPURL:
-			urlParts[key] = value.Str()
+		case conventions.AttributeHTTPURL, AttributeURLFull:
+			urlParts[conventions.AttributeHTTPURL] = value.Str()
 			hasHTTP = true
 			hasHTTPRequestURLAttributes = true
-		case conventions.AttributeHTTPScheme:
-			urlParts[key] = value.Str()
+		case conventions.AttributeHTTPScheme, AttributeURLSchema:
+			urlParts[conventions.AttributeHTTPScheme] = value.Str()
 			hasHTTP = true
-		case conventions.AttributeHTTPHost:
-			urlParts[key] = value.Str()
+		case conventions.AttributeHTTPHost, AttributeServerAddress:
+			urlParts[conventions.AttributeHTTPHost] = value.Str()
 			hasHTTP = true
 			hasHTTPRequestURLAttributes = true
-		case conventions.AttributeHTTPTarget:
-			urlParts[key] = value.Str()
+		case conventions.AttributeHTTPTarget, AttributeURLPath:
+			urlParts[conventions.AttributeHTTPTarget] = value.Str()
 			hasHTTP = true
-		case conventions.AttributeHTTPServerName:
+		case conventions.AttributeHTTPServerName: // Deprecated, use `server.address`
 			urlParts[key] = value.Str()
 			hasHTTP = true
 			hasHTTPRequestURLAttributes = true
-		case conventions.AttributeNetHostPort:
-			urlParts[key] = value.Str()
+		case conventions.AttributeNetHostPort, AttributeServerPort:
+			urlParts[conventions.AttributeNetHostPort] = value.Str()
 			hasHTTP = true
-			if len(urlParts[key]) == 0 {
-				urlParts[key] = strconv.FormatInt(value.Int(), 10)
+			if len(urlParts[conventions.AttributeNetHostPort]) == 0 {
+				urlParts[conventions.AttributeNetHostPort] = strconv.FormatInt(value.Int(), 10)
 			}
-		case conventions.AttributeHostName:
+		case conventions.AttributeHostName: // Deprecated, use `server.address`
 			urlParts[key] = value.Str()
 			hasHTTPRequestURLAttributes = true
-		case conventions.AttributeNetHostName:
+		case conventions.AttributeNetHostName: // Deprecated, use `server.address`
 			urlParts[key] = value.Str()
 			hasHTTPRequestURLAttributes = true
-		case conventions.AttributeNetPeerName:
-			urlParts[key] = value.Str()
-		case conventions.AttributeNetPeerPort:
+		case conventions.AttributeNetPeerName: // Deprecated, use `client.address`
 			urlParts[key] = value.Str()
-			if len(urlParts[key]) == 0 {
+		case conventions.AttributeNetPeerPort, AttributeClientPort:
```

Comment:
`net.peer.port` is deprecated in v1.23.0, use `server.address` on client spans and `client.address` on server spans. That means if span has attribute `client.address` we can follow the existing 'net.peer.port' translation logic.

This case no need to cover `server.address` because it is already covered by another case:
```
case conventions.AttributeHTTPHost, AttributeServerAddress:
			urlParts[conventions.AttributeHTTPHost] = value.Str()
			hasHTTP = true
			hasHTTPRequestURLAttributes = true
```

Reply:
So, I don't think `*.address` is relevant, you probably meant `*.port`. In any case, [here's where `net.peer.port` was deprecated](https://github.com/open-telemetry/opentelemetry-specification/pull/3402/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR63-R65), this is telling us that `net.peer.port` was deprecated in favor of two attributes `server.port` and `client.port`:

> BREAKING: rename `net.peer.name` to `server.address` on client side and to `client.address` on server side, `net.peer.port` to `server.port` on client side and to `client.port` on server side, `net.host.name` and `net.host.port` to `server.name` and `server.port` (since `net.host.*` attributes only applied to server instrumentation)

So in a few cases, we are splitting one attribute into two. What this means is if we were once looking for `net.peer.port`, we now need to look for both `server.port` and `client.port`, unless we have some way of distinguishing if we are in the server or client perspective.

Now, you do have a case for `server.port`, but that doesn't do the same thing:

```
case conventions.AttributeNetHostPort, AttributeServerPort:
  urlParts[conventions.AttributeNetHostPort] = value.Str()
  hasHTTP = true
  if len(urlParts[conventions.AttributeNetHostPort]) == 0 {
    urlParts[conventions.AttributeNetHostPort] = strconv.FormatInt(value.Int(), 10)
  }
```

vs

```
case conventions.AttributeNetPeerPort, AttributeClientPort:
  urlParts[conventions.AttributeNetPeerPort] = value.Str()
  if len(urlParts[conventions.AttributeNetPeerPort]) == 0 {
    urlParts[key] = strconv.FormatInt(value.Int(), 10)
  }
```

What this seems to suggest is that if we see either AttributeServerPort or AttributeClientPort, we need to run the AttributeNetPeerPort logic, and if we see AttributeServerPort, we need to also run the AttributeNetHostPort logic.

Similar logic follows for `net.peer.name`

Reply:
We are lucky that `url.full` is required attribute in Http client span, so we don't have to build http url for http client span by `client.address` and `client.port`.
Only `server.address` and `server.port` are needed for building http url for http server span.